src/vs/workbench/contrib/notebook/browser/controller/editActions.ts

```diff
@@ -157,6 +160,37 @@ registerAction2(class DeleteCellAction extends NotebookCellAction {
 			return;
 		}
 
+		let confirmation: IConfirmationResult;
+		const notebookExecutionStateService = accessor.get(INotebookExecutionStateService);
+		const runState = notebookExecutionStateService.getCellExecution(context.cell.uri)?.state;
+		const configService = accessor.get(IConfigurationService);
+
+		if (runState === NotebookCellExecutionState.Executing && configService.getValue(NotebookSetting.confirmDeleteRunningCell)) {
+			const dialogService = accessor.get(IDialogService);
+			const primaryButton = localize('confirmDeleteButton', 'Confirm Delete');
+
+			confirmation = await dialogService.confirm({
+				type: 'question',
+				message: localize('confirmDeleteButtonMessage', 'Are you sure you want to delete a running cell?'),
+				detail: localize('confirmDeleteButtonDetail', 'After deletion, the cell may not be recovered.'),
```

Comment:
That's not true, "undo" exists

Reply:
I deleted a cell while it was running and undo didn't seem to take effect, so do we need to delete the description of the detail

Reply:
Works for me- that sounds like it would be a separate issue.

Reply:
You can delete the detail, and everything else here looks fine